kai/reactive_codeplanner/task_runner/compiler/compiler_task_runner.py

```diff
@@ -14,6 +14,8 @@
 from kai.reactive_codeplanner.task_runner.compiler.maven_validator import (
     AccessControlError,
     AnnotationError,
+    BuildError,
+    DependencyResolutionError,
```

Comment:
@shawn-hurley Should this be handled in the dependency resolution instead? This is the error it's detecting:


```python
[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< com.redhat.coolstore:monolith >--------------------
[INFO] Building coolstore-monolith 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from github: https://maven.pkg.github.com/shawn-hurley/analyzer-lsp/jakarta/jakarta.jakartaee-web-api/8.0.0/jakarta.jakartaee-web-api-8.0.0.pom
Downloading from github: https://maven.pkg.github.com/shawn-hurley/analyzer-lsp/jakarta/jakarta.jakartaee-api/8.0.0/jakarta.jakartaee-api-8.0.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.749 s
[INFO] Finished at: 2024-11-13T20:07:09-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project monolith: Could not resolve dependencies for project com.redhat.coolstore:monolith:jar:1.0.0-SNAPSHOT: Failed to collect dependencies at jakarta:jakarta.jakartaee-web-api:jar:8.0.0: Failed to read artif
act descriptor for jakarta:jakarta.jakartaee-web-api:jar:8.0.0: The following artifacts could not be resolved: jakarta:jakarta.jakartaee-web-api:pom:8.0.0 (absent): Could not transfer artifact jakarta:jakarta.jakartaee-web-api:pom:8.0.0 from/to
 github (https://maven.pkg.github.com/shawn-hurley/analyzer-lsp): status code: 401, reason phrase: Unauthorized (401) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

[INFO] Scanning for projects...
[INFO]
[INFO] -------------------< com.redhat.coolstore:monolith >--------------------
[INFO] Building coolstore-monolith 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from github: https://maven.pkg.github.com/shawn-hurley/analyzer-lsp/jakarta/jakarta.jakartaee-web-api/8.0.0/jakarta.jakartaee-web-api-8.0.0.pom
Downloading from github: https://maven.pkg.github.com/shawn-hurley/analyzer-lsp/jakarta/jakarta.jakartaee-api/8.0.0/jakarta.jakartaee-api-8.0.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.749 s
[INFO] Finished at: 2024-11-13T20:07:09-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project monolith: Could not resolve dependencies for project com.redhat.coolstore:monolith:jar:1.0.0-SNAPSHOT: Failed to collect dependencies at jakarta:jakarta.jakartaee-web-api:jar:8.0.0: Failed to read artif
act descriptor for jakarta:jakarta.jakartaee-web-api:jar:8.0.0: The following artifacts could not be resolved: jakarta:jakarta.jakartaee-web-api:pom:8.0.0 (absent): Could not transfer artifact jakarta:jakarta.jakartaee-web-api:pom:8.0.0 from/to
 github (https://maven.pkg.github.com/shawn-hurley/analyzer-lsp): status code: 401, reason phrase: Unauthorized (401) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

Reply:
Neither of them seem capable of solving the issue at the moment

Reply:
How is it writing my github into the maven package? what is happening!

In all seriousness, the FQDN agent should be able to solve this with some tweaking, which may be worth looking into.